R-package/R/xgb.DMatrix.R

```diff
@@ -18,11 +46,32 @@
 #' @param base_margin Base margin used for boosting from existing model.
 #'
 #'        In the case of multi-output models, one can also pass multi-dimensional base_margin.
-#' @param missing a float value to represents missing values in data (used only when input is a dense matrix).
-#'        It is useful when a 0 or some other extreme value represents missing values in data.
+#' @param missing a float value to represents missing values in data (not used when creating DMatrix
+#'        from text files).
+#'        It is useful to change when a zero, infinite, or some other extreme value represents missing
+#'        values in data.
 #' @param silent whether to suppress printing an informational message after loading from a file.
 #' @param feature_names Set names for features. Overrides column names in data
 #'        frame and matrix.
+#'
+#'        Note: columns are not referenced by name when calling `predict`, so the column order there
```

Comment:
I think this is an area that I should work on, to make sure the ordering is correct, at least for dataframe.

Reply:
I just remember the `valid_features` parameter in Python, there's a check in there to verify the columns are consistent.

Reply:
I think this sort of check should definitely not be enabled by default in a prediction function, since it can potentially add a lot of latency when serving a model.

My idea was to have such a check implemented in the new `xgboost()` interface (with a separate class for the model object), but would be ideal if it could be done in C++ without having to copy the names to R character vectors.

That being said, in e.g. R's `predict.glm`, if the data contains the column names but in a different order, then the input data would get re-ordered instead of throwing an error.

Reply:
The check in Python is optional but enabled by default. I'm not sure how to do it in the new `xgboost` interface though since the check is for inference while the `xgboost()` function is for training.

Reply:
My idea was to make the output from `xgboost()` be of a different class than `xgb.train`, just like the python interface has scikit-learn classes which are different from `xgb.Booster` - then, this class coming out of `xgboost()` would have a different `predict` method which could do extra things, such as add class names, extra checks, etc.

Reply:
Although, if the data has column names and so does the booster, it should be possible to add an extra parameter in `predict.xgb.Booster` to optionally make it behave like R's `predict.glm` which reorders columns and fails if they aren't present.

Reply:
Got it, we will do the check in your way.